src/main/kotlin/org/jetbrains/research/ml/codetracker/reporting/ErrorReport.kt

```diff
@@ -35,19 +39,20 @@ object ErrorReport {
                 issueService,
                 repoID
             )
-            val isNewIssue = duplicate?.let {
+            var isNewIssue = false
+            if (duplicate != null) {
                 val newErrorComment: String = generateGitHubIssueBody(errorInformation)
-                issueService.createComment(repoID, it.number, newErrorComment)
-                newGibHubIssue = it
-                false
-            } ?: {
+                issueService.createComment(repoID, duplicate.number, newErrorComment)
+                newGibHubIssue = duplicate
+            } else {
                 newGibHubIssue = issueService.createIssue(repoID, newGibHubIssue)
-                true
-            }()
+                isNewIssue = true
+            }
+            val submissionStatus = if (isNewIssue) SubmissionStatus.NEW_ISSUE else SubmissionStatus.DUPLICATE
```

Comment:
Why don't you want to init submissionStatus in if-else block that you already have (with checking if duplicate is null)? So you don't need to create an additional variable with isNewIssue

Reply:
Nice idea, thanks!